src/Symfony/Component/Serializer/Tests/Encoder/CsvEncoderTest.php

```diff
@@ -710,4 +707,13 @@ public function testEndOfLinePassedInConstructor()
         $encoder = new CsvEncoder([CsvEncoder::END_OF_LINE => "\r\n"]);
         $this->assertSame("foo,bar\r\nhello,test\r\n", $encoder->encode($value, 'csv'));
     }
+
+    /**
+     * @group legacy
+     */
+    public function testPassingAnNonEmptyEscapeCharIsDeprecated()
+    {
+        $this->expectDeprecation('Since symfony/serializer 7.2: Setting the "csv_escape_char" option is deprecated. The option will be removed in 8.0.');
+        new CsvEncoder(['csv_escape_char' => '\\']);
```

Comment:
shouldn't we also have an assertion ensuring that the behavior of the escape char keeps working when it is passed in the options ?

Reply:
Indeed, here you go